File: backend/src/main/java/com/tenga/auth/model/dto/AuthTokenResponse.java
package com.tenga.auth.model.dto;

import com.tenga.auth.model.enums.UserRole;

public record AuthTokenResponse(
    String accessToken, String tokenType, long expiresIn, UserRole role, String userId) {

  public static AuthTokenResponse of(
      String accessToken, long expiresIn, UserRole role, String userId) {
    return new AuthTokenResponse(accessToken, "Bearer", expiresIn, role, userId);
  }
}

File: mobile/lib/features/auth/data/models/auth_token_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'auth_token_response.freezed.dart';
part 'auth_token_response.g.dart';

@freezed
class AuthTokenResponse with _$AuthTokenResponse {
  const factory AuthTokenResponse({
    required String accessToken,
    required String tokenType,
    required int expiresIn,
    required String role,
    required String userId,
  }) = _AuthTokenResponse;

  factory AuthTokenResponse.fromJson(Map<String, dynamic> json) =>
      _$AuthTokenResponseFromJson(json);
}
